<?xml version="1.0"?>
<launch>
    
  <rosparam file="$(find three_dof_planar_manipulator)/config/controllers.yaml" command="load"/>
   
  <arg name="model" default="$(find three_dof_planar_manipulator)/urdf/three_dof_planar_manipulator.urdf.xacro"/>
  <arg name="gui" default="true" />
  
  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/>

  <node name="robot_hardware_interface" pkg="three_dof_planar_manipulator" type="robot_hardware_interface" output="screen"/>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" >
  </node>
  
  <node name="rviz" pkg="rviz" type="rviz"/>
    
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="
        	/three_dof_planar_manipulator/joints_update
            /three_dof_planar_manipulator/joint1
            /three_dof_planar_manipulator/joint2
            /three_dof_planar_manipulator/joint3
        "/>
</launch>
